src/contract_clients/core_contract.rs

```diff
@@ -0,0 +1,283 @@
+use std::future::Future;
+use std::sync::Arc;
+
+use async_trait::async_trait;
+use ethers::abi::AbiEncode;
+use ethers::addressbook::Address;
+use ethers::prelude::{Bytes, I256, U256};
+use starknet_api::hash::{StarkFelt, StarkHash};
+use starknet_ff::FieldElement;
+use starknet_proxy_client::proxy_support::{CoreContractInitData, CoreContractState};
+use zaun_utils::LocalWalletSignerMiddleware;
+
+use crate::contract_clients::config::Config;
+use crate::contract_clients::starknet_sovereign::StarknetSovereignContract;
+use crate::contract_clients::starknet_validity::StarknetValidityContract;
+use crate::utils::convert_felt_to_u256;
+
+pub enum CoreContractStarknetL1Enum {
```

Comment:
Do we really need this enum? Can we not implement trait `CoreContract` for `StarknetSovereignContract` and `StarknetValidityContract` and when we call those methods we call them with `Box<dyn CoreContract>`

Reply:
can you give an example of this ? I didn't get it I added dyn CoreContract wherever required.

Reply:
was this fixed, looks like it was

Reply:
yes it is fixed now